docs/graphs/Adjacency-Matrix.md

```diff
@@ -0,0 +1,118 @@
+---
+id: adjacency-matrix
+title: Adjacency matrix
+sidebar_label: Adjacency matrix
+description: "An adjacency matrix is a 2D array used to represent a graph, where each cell (i, j) is set to 1 if there's an edge from node i to node j, and 0 otherwise."
+tags: [Graph,Algorithm, DSA]
+---
+# Adjacency Matrix
+## Problem Statement
+Write a program in to create adjacency matrix of a given graph. If a graph has n vertices, we use n x n matrix to represent the graph.Let's assume the n x n matrix as adj[n][n].
+- if there is an edge from vertex i to j, mark adj[i][j] as 1. i.e. adj[i][j] == 1
+- if there is no edge from vertex i to j, mark adj[i][j] as 0. i.e. adj[i][j] == 0
+
+## Approach:
+The approach constructs a graph using an adjacency matrix, initialized to 0, where each user-specified neighbor for a node sets the corresponding matrix entry to 1, visually representing edges between nodes when displayed.
+
+## Algorithm Overview
+
+1. **Input the Number of Nodes**: Prompt the user to input the number of nodes in the graph. Initialize an adjacency matrix of size  `V x V`  with all entries set to zero.
+
+2. **Create the Adjacency Matrix**:
+   - For each node, ask the user for the number of neighbors.
+   - For each neighbor, set the corresponding matrix entry to 1 to indicate an edge between the nodes.
+
+3. **Display the Adjacency Matrix**:
+   - Print a formatted adjacency matrix that shows connections between nodes, with rows and columns representing nodes and entries indicating edges.
+
+4. **User Interaction**:
+   - The program takes user input for each node’s neighbors, allowing dynamic graph construction based on user-defined connections.
+
+## Example
+
+### Sample Input:
+ 
+Enter the number of nodes in G: 5                  
+
+Enter the number of neighbors of 0: 2                               
+Enter the neighbors of 0 (0-based indices): 1 2                         
+
+Enter the number of neighbors of 1: 2                                    
+Enter the neighbors of 1 (0-based indices): 0 3                           
+
+Enter the number of neighbors of 2: 2                      
+Enter the neighbors of 2 (0-based indices): 0 3                           
+
+Enter the number of neighbors of 3: 3                          
+Enter the neighbors of 3 (0-based indices): 2 1 4                        
+
+Enter the number of neighbors of 4: 1                       
+Enter the neighbors of 4 (0-based indices): 3                      
+
+```
+   0 ---- 1
+   |      |
+   2 ---- 1 --- 4                       
+```
```

Comment:
@ajay-dhangar Sorry to say you that in this part of line I have made a very littele mistake that is the connecting edge with 2 will be 3  i.e 
```
   0 ---- 1
   |      |
   2 ---- 3 --- 4                       
``` 
As you have already merged please edit that 1 with 3.